### PR TITLE
show_header_names decoration

### LINKED ISSUE #1696: `show_header_names()` updates
- Do we still need the `include_example` argument?
- The table is currently being printed with kable (aka a markdown table), if my memory is correct. I want to update this to be a {cli} print that is more beautiful, highlighting the parts that are most important, useful for users.
- I also want to print the stats that are available to dynamically include in the headers (see example below), along with their types.

``` r
library(gtsummary)
packageVersion("gtsummary")
#> [1] '1.7.2'

tbl <- tbl_summary(
  trial,
  by = trt, 
  include = age
) |> 
  add_overall()

show_header_names(tbl)
#> ℹ As a usage guide, the code below re-creates the current column headers.
#> modify_header(
#>   label = '**Characteristic**',
#>   stat_0 = '**Overall**, N = 200',
#>   stat_1 = '**Drug A**, N = 98',
#>   stat_2 = '**Drug B**, N = 102'
#> )
#> 
#> 
#> Column Name   Column Header        
#> ------------  ---------------------
#> label         **Characteristic**   
#> stat_0        **Overall**, N = 200 
#> stat_1        **Drug A**, N = 98   
#> stat_2        **Drug B**, N = 102

# i also want to be able to display what stats are available to report in the headers
#   AND the type (e.g. <int>, <dbl>, <chr>, etc.)
tbl$table_styling$header |> 
  dplyr::select(column, label, starts_with("modify_stat_")) |> 
  dplyr::rename_with(.fn = ~gsub("^modify_stat_", "", .x))
#> # A tibble: 8 × 6
#>   column    label                    N     n     p level  
#>   <chr>     <chr>                <int> <int> <dbl> <chr>  
#> 1 variable  variable               200    NA NA    <NA>   
#> 2 var_type  var_type               200    NA NA    <NA>   
#> 3 var_label var_label              200    NA NA    <NA>   
#> 4 row_type  row_type               200    NA NA    <NA>   
#> 5 label     **Characteristic**     200    NA NA    <NA>   
#> 6 stat_0    **Overall**, N = 200   200   200  1    Overall
#> 7 stat_1    **Drug A**, N = 98     200    98  0.49 Drug A 
#> 8 stat_2    **Drug B**, N = 102    200   102  0.51 Drug B
```

<sup>Created on 2024-05-30 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by ayogasekaram**
**What changes are proposed in this pull request?**
- change header text colour
- change dynamic value text colour
- add data type of the dynamic values and italicize
- replace NA vals with ""

closes #1696 

**If there is an GitHub issue associated with this pull request, please provide link.**

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] Ensure all package dependencies are installed: `renv::install()`
- [ ] PR branch has pulled the most recent updates from master branch: `usethis::pr_merge_main()`
- [ ] If a bug was fixed, a unit test was added.
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features: `devtools::test_coverage()`
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: R/modify.R
Comment (ddsjoberg) [reply R/modify.R:283]: Can we add a test for a class that we don't have specific abbreviations for?

**Review by ddsjoberg — CHANGES_REQUESTED** [review]
@ayogasekaram thank you so much for this update! I love it!! A few comments to address before merging

File: R/tibble_as_cli.R
Comment (ayogasekaram) [reply R/tibble_as_cli.R:94]: I tried using ansi_strip in the padding parts of the function but it then loses any of the ansi code that would be added from the modify function first.

File: tests/testthat/_snaps/show_header_names.md
Comment (ayogasekaram) [reply tests/testthat/_snaps/show_header_names.md]: Great catch Thank you Daniel! This has been updated in the show_header_names code

File: R/tibble_as_cli.R
Comment (ddsjoberg) [reply R/tibble_as_cli.R:93]: ok, thank you for looking into it!

File: R/modify.R
Comment (ayogasekaram) [reply R/modify.R]: I've updated the function so it evaluates the return of the `class(.)` section and maps the first "valid" class. So if it returned a vector, it would only paste 1 value in the <>

**Comment by ddsjoberg** [comment]
Sorry it took me so long to get to the final review!

It looks like the code that rounds the stats appeared after the class abbreviation was appended, which forced the column to character. I'll re-arrange the calls a bit and push. Thank you!!

image

**Comment by ddsjoberg** [comment]
Thank you @ayogasekaram ! The results look good to me. Can you double check that my edits didn't break some of your existing code?

**Comment by ayogasekaram** [comment]
> Thank you @ayogasekaram ! The results look good to me. Can you double check that my edits didn't break some of your existing code?

The tests are passing the the code looks great! Thank you @ddsjoberg!

**Pull request merged**
